plan/logical_plans.go

```diff
@@ -268,6 +268,9 @@ type DataSource struct {
 	pushedDownConds []expression.Expression
 
 	statisticTable *statistics.Table
+
+	// NeedColHandle is used in execution phase.
+	NeedColHandle bool
```

Comment:
We can maintain a `HandleOffset` instead of NeedColHandle. If the PK column exists in table's columns, we needn't append an extra column.